### PR TITLE
[toggle]Aria-expanded docs and example added

### LINKED ISSUE #1920: Document using aria attributes with element used to toggle terra-toggle component
# Feature Request

## Description
Certain aria attributes should be used when expanding and collapsing the terra-toggle component. These need to be set on the element users are interacting with outside of the terra-toggle component that is triggering the expand and collapse. We should document these so they are clear for our users.

https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Fixes #1920

### CODE REVIEW

File: packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx [hunk 1]
```diff
@@ -19,7 +19,10 @@ class ToggleDefault extends React.Component {
     const { isOpen } = this.state;
     return (
       <div>
-        <IconInformation onClick={this.handleOnClick} />
+        <IconInformation onClick={this.handleOnClick} aria-expanded={isOpen} />
```
Comment (bjankord) [inline packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx]: Let's add the aria-controls attribute as well
```suggestion
        <IconInformation onClick={this.handleOnClick} aria-expanded={isOpen} aria-controls="toggleContainer" />
```

File: packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx [hunk 2]
```diff
@@ -19,7 +19,10 @@ class ToggleDefault extends React.Component {
...
+        {/**
+        * The aria-expanded state is used on the triggering component to indicate the contents are collapsible, and whether a region is currently expanded or collapsed
+        */}
         <Toggle isOpen={isOpen} isAnimated>
```
Comment (bjankord) [inline packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx]: Let's add the id for the aria-controls mapping:
```suggestion
        <Toggle id="toggleContainer" isOpen={isOpen} isAnimated>
```

File: packages/terra-toggle/src/terra-dev-site/doc/example/DefaultToggle.jsx [hunk 3]
```diff
@@ -19,7 +19,10 @@ class ToggleDefault extends React.Component {
     const { isOpen } = this.state;
     return (
       <div>
-        <IconInformation onClick={this.handleOnClick} />
+        <IconInformation onClick={this.handleOnClick} aria-expanded={isOpen} />
```
Comment (bjankord) [inline packages/terra-toggle/src/terra-dev-site/doc/example/DefaultToggle.jsx]: Let's add the aria-controls and id similar to `packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx`

File: packages/terra-toggle/src/terra-dev-site/test/toggle/AnimatedToggle.test.jsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ class AnimatedToggle extends React.Component {
     const { isOpen } = this.state;
     return (
       <div>
-        <button type="button" id="trigger-toggle" onClick={this.handleOnClick}>Animated Toggle</button>
+        <button type="button" id="trigger-toggle" onClick={this.handleOnClick} aria-expanded={isOpen}>Animated Toggle</button>
```
Comment (bjankord) [inline packages/terra-toggle/src/terra-dev-site/test/toggle/AnimatedToggle.test.jsx]: Let's add the aria-controls and id similar to `packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx`

File: packages/terra-toggle/src/terra-dev-site/test/toggle/DefaultToggle.test.jsx [hunk 5]
```diff
@@ -14,9 +14,10 @@ class ToggleDefault extends React.Component {
   }
 
   render() {
+    const { isOpen } = this.state;
     return (
       <div>
-        <button type="button" id="trigger-toggle" onClick={this.handleOnClick}>Default Toggle</button>
+        <button type="button" id="trigger-toggle" onClick={this.handleOnClick} aria-expanded={isOpen}>Default Toggle</button>
```
Comment (bjankord) [inline packages/terra-toggle/src/terra-dev-site/test/toggle/DefaultToggle.test.jsx]: Let's add the aria-controls and id similar to `packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx`

File: packages/terra-toggle/src/terra-dev-site/test/toggle/OpenToggle.test.jsx [hunk 6]
```diff
@@ -17,7 +17,7 @@ class OpenToggle extends React.Component {
     const { isOpen } = this.state;
     return (
       <div>
-        <button type="button" id="trigger-toggle" onClick={this.handleOnClick}>Open Toggle</button>
+        <button type="button" id="trigger-toggle" onClick={this.handleOnClick} aria-expanded={isOpen}>Open Toggle</button>
```
Comment (bjankord) [inline packages/terra-toggle/src/terra-dev-site/test/toggle/OpenToggle.test.jsx]: Let's add the aria-controls and id similar to `packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx`

File: packages/terra-toggle/docs/README.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Terra Toggle
 
-Toggle component that transitions content in and out. For toggle button functionality, see [terra-toggle-button](https://engineering.cerner.com/terra-ui/#/components/terra-toggle-button/toggle-button/toggle-button).
+Toggle component that transitions content in and out. Use aria-expanded state on the triggering component. The objective of aria-expanded state is to indicate whether regions of the content are collapsible, and to expose whether a region is currently expanded or collapsed. For toggle button functionality, see [terra-toggle-button](https://engineering.cerner.com/terra-ui/#/components/terra-toggle-button/toggle-button/toggle-button).
```
Comment (bjankord) [inline packages/terra-toggle/docs/README.md]: Let's add a note on using aria-controls as well like the example from https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions#Example_1:_Using_a_button_to_collapse_and_expand_a_region

**Comment by supreethmr** [comment]
Tested in VO. VO is not reading collapsed and expanded state on click of information icon on dev-site examples. But VO is working as expected for test examples which has button for toggle. I feel we may need to check this difference in behaviour of button and Icon

File: packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx (hunk 1 above)
Comment (saket2403) [reply packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx]: Resolved in 3478f79

File: packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx (hunk 2 above)
Comment (saket2403) [reply packages/terra-toggle/src/terra-dev-site/doc/example/AnimatedToggle.jsx]: Resolved in 3478f79

File: packages/terra-toggle/src/terra-dev-site/doc/example/DefaultToggle.jsx (hunk 3 above)
Comment (saket2403) [reply packages/terra-toggle/src/terra-dev-site/doc/example/DefaultToggle.jsx]: Resolved in 3478f79

File: packages/terra-toggle/src/terra-dev-site/test/toggle/AnimatedToggle.test.jsx (hunk 4 above)
Comment (saket2403) [reply packages/terra-toggle/src/terra-dev-site/test/toggle/AnimatedToggle.test.jsx]: Resolved in 3478f79

File: packages/terra-toggle/src/terra-dev-site/test/toggle/DefaultToggle.test.jsx (hunk 5 above)
Comment (saket2403) [reply packages/terra-toggle/src/terra-dev-site/test/toggle/DefaultToggle.test.jsx]: Resolved in 3478f79

File: packages/terra-toggle/docs/README.md (hunk 7 above)
Comment (saket2403) [reply packages/terra-toggle/docs/README.md]: Resolved in 3478f79

File: packages/terra-toggle/docs/README.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # Terra Toggle
 
-Toggle component that transitions content in and out. For toggle button functionality, see [terra-toggle-button](https://engineering.cerner.com/terra-ui/#/components/terra-toggle-button/toggle-button/toggle-button).
+Toggle component that transitions content in and out. Use aria-expanded and aria-controls  state on the triggering component. The objective of aria-expanded state is to indicate whether regions of the content are collapsible, and to expose whether a region is currently expanded or collapsed.The aria-controls state indicates the element (terra-toggle) that is controlled by the triggering element. For toggle button functionality, see [terra-toggle-button](https://engineering.cerner.com/terra-ui/#/components/terra-toggle-button/toggle-button/toggle-button).
```
Comment (yuderekyu) [inline packages/terra-toggle/docs/README.md]:
```suggestion
Toggle component that transitions content in and out. Use `aria-expanded` and `aria-controls` state on the triggering component. The objective of `aria-expanded` state is to indicate whether regions of the content are collapsible, and to expose whether a region is currently expanded or collapsed.The `aria-controls` state indicates the element (terra-toggle) that is controlled by the triggering element. For toggle button functionality, see [terra-toggle-button](https://engineering.cerner.com/terra-ui/#/components/terra-toggle-button/toggle-button/toggle-button).
```

Comment (saket2403) [reply packages/terra-toggle/docs/README.md]: Resolved in 82aa7a8

File: packages/terra-toggle/CHANGELOG.md [hunk 9]
```diff
@@ -6,10 +6,12 @@ Unreleased
 ### Changed
 * updated package.json test scripts
 
-
 ### Changed
```
Comment (avinashg1994) [inline packages/terra-toggle/CHANGELOG.md]: nit: Can you combine the two "Changed" sections into a single section

**Pull request merged**